rollup.config.js

```diff
@@ -166,7 +175,9 @@ module.exports = function (args) {
   }
   return [
     ...(c === 'index' ? [createDeclarationConfig(`src/${c}.ts`, 'dist')] : []),
-    createCommonJSConfig(`src/${c}.ts`, `dist/${c}`),
+    createCommonJSConfig(`src/${c}.ts`, `dist/${c}`, {
+      addModuleExport: c === 'index',
```

Comment:
We want this for `zustand`, `zustand/vanilla`, `zustand/shallow` and `zustand/shallow`.

Reply:
for some reason the `zustand/{vanilla,shallow}` already exports itself as `module.exports` in the build so the patch wasn't needed, also why I decided to go with a selective way to add this line. 

I do realise I have to add this to middleware but I wanted to check if it's working before I add that

Reply:
> for some reason the `zustand/{vanilla,shallow}` already exports itself as `module.exports` in the build

https://unpkg.com/browse/zustand@4.1.4/vanilla.js
Oh, I didn't know that...
Maybe because there are no named exports.

> I do realise I have to add this to middleware

We don't have default exports in `zustand/middleware`, so it's not necessary.